### PR TITLE
#O21KB-2 Front-End: Sass Variables, Mixins, Breakpoints, Colors and Font-face

### PR DESCRIPTION
**Pull request by nilouhou**
Created global sass for instock project based on https://www.figma.com/file/oRk4LqCj1wYYEtfqZwzdDX/InStock-Mockups?node-id=1196%3A351

** What is included:**

- Typography
- Colors
- Layout
- Breakpoints
- reset

Feel free to mention any elements that should be included in global styles sheet

### CODE REVIEW

**Comment by FadyaJoudah** [comment]
Nilou how do you feel about doing the request against the develop? 
should we leave that and merge straight to main?

File: client/src/styles/partials/_mixins.scss [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+@mixin flex-column-center {
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+}
+
+@mixin flex-row {
+	display: flex;
+	flex-direction: row;
```
Comment (manecipoglu) [inline client/src/styles/partials/_mixins.scss:19]: I believe flex-direction: row is the default behavior. Do you want to keep it for being explicit?

File: client/src/styles/partials/_typography.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* Fonts */
+
+/** Regular **/
+
+@font-face {
+	font-family: "TitilliumWeb";
+	src: url("../../assets/fonts/TitilliumWeb-Regular.ttf") format("truetype");
```
Comment (manecipoglu) [inline client/src/styles/partials/_typography.scss:7]: Maybe it is a good idea to transform the font to a woff2 format for better performance.

**Comment by nilouhou** [comment]
> Nilou how do you feel about doing the request against the develop? should we leave that and merge straight to main?

Thanks Fadya, changed the base

File: client/src/styles/partials/_mixins.scss (hunk 1 above)
Comment (nilouhou) [reply client/src/styles/partials/_mixins.scss:19]: yes, agree, will fix that

File: client/src/styles/partials/_typography.scss (hunk 2 above)
Comment (nilouhou) [reply client/src/styles/partials/_typography.scss:7]: Like, keep this, one and convert to woff2, too?

Comment (manecipoglu) [reply client/src/styles/partials/_typography.scss:7]: I would just have the .woff2 but one might argue to have both and use .ttf one as the fallback.

Comment (shaynelar) [reply client/src/styles/partials/_typography.scss:7]: I think what Muammar said here makes sense

Comment (nilouhou) [reply client/src/styles/partials/_typography.scss:7]: sure, I add woff2 as main woff and ttf for fallback

File: client/src/styles/partials/_mixins.scss (hunk 1 above)
Comment (nilouhou) [reply client/src/styles/partials/_mixins.scss]: fixed, please review

File: client/src/styles/partials/_typography.scss (hunk 2 above)
Comment (nilouhou) [reply client/src/styles/partials/_typography.scss]: fixed, please review

**Comment by manecipoglu** [comment]
Looks awesome Nilou! Thank you. Your setup is much deeper than what I have been using and I will take a look a few times to remember the tools you made for us to use.

**Comment by nilouhou** [comment]
@manecipoglu  thank you so much for your kind words

File: client/src/styles/partials/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+/* Supporting Colours */
+$success-green: #158463;
+$reject-red: #c94515;
+
+/* Break points */
+$mobile-breakpoint: 320px;
+$tablet-breakpoint: 768px;
+$desktop-breakpoint: 1280px;
+
```
Comment (shaynelar) [inline client/src/styles/partials/_variables.scss:26]: Breakpoints should probably be rem for consistency

Comment (nilouhou) [reply client/src/styles/partials/_variables.scss:26]: I understand your point of view, but I think pixel for media query might be safer choice ex. Safari

**Pull request merged**